src/sql/workbench/services/connection/browser/connectionDialogWidget.ts

```diff
@@ -265,6 +272,7 @@ export class ConnectionDialogWidget extends Modal {
 		this.browsePanel = new ConnectionBrowseTab(this.instantiationService);
 
 		this._register(this.browsePanel.view.onSelectedConnectionChanged(e => {
+			this._connectionSource = e.source;
```

Comment:
What if the person clicks a connection to select it but then modifies the populated values in some way?

Reply:
Yeah, I did think about that, I think it is acceptable since the user actually used those features to begin with.